docs/topics/yaml-rule-structure-syntax.adoc

```diff
@@ -0,0 +1,520 @@
+
+[source,terminal]
+----
+when:
+  <condition>
+    <nested-condition>
+----
+
+{ProductShortName} supports three types of conditions: `provider`, `and` and `or`.
+
+==== Provider conditions
+
+{ProductShortName} supports multi-language source code analysis. Source code search for a specific language is enabled using the `provider` condition. This condition defines a search query for a specific language provider.  The `provider` condition also specifies which of the provider's "capabilities" to use for analyzing the code.
+
+The `provider` condition has the form `<provider_name>.<capability>`:
+
+[source,terminal]
+----
+when:
+  <provider_name>.<capability>
+    <input_fields>
+----
+
+The analyzer currently supports the following `provider` conditions:
+
+* `builtin`
+* `java`
+* `go`
+
+===== `builtin` provider
+
+`builtin` is an internal provider that can analyze various files and internal metadata generated by the engine.
+
+This provider has the following capabilities:
+
+* `file`
+* `filecontent`
+* `xml`
+* `json`
+* `hasTags`
+
+.`file`
+
+The `file` capability enables the provider to search for files in the source code that match a given pattern.
+
+[source,terminal]
+----
+when:
+  builtin.file:
+    pattern: "<regex_to_match_filenames>"
+----
+
+.`filecontent`
+
+The `filecontent` capability enables the provider to search for content that matches a given pattern.
+
+[source,terminal]
+----
+when:
+  builtin.filecontent:
+    filePattern: "<regex_to_match_filenames_to_scope_search>"
+    pattern: "<regex_to_match_content_in_the_matching_files>"
+----
+
+.`xml`
+
+The `xml` capability enables the provider to query XPath expressions on a list of provided XML files. This capability takes 2 input parameters.
+
+[source,terminal]
+----
+when:
+  builtin.xml:
+    xpath: "<xpath_expressions>" <1>
+    filepaths: <2>
+      - "/src/file1.xml"
+      - "/src/file2.xml"
+----
+<1> `xpath` must be a valid XPath expression.
+<2> `filepaths` is a list of files to apply the XPath query to.
+
+.`json`
+
+The `json` capability enables the provider to query XPath expressions on a list of provided JSON files. Currently, `json` only takes XPath as input and performs the search on all JSON files in the codebase.
+
+[source,terminal]
+----
+when:
+  builtin.json:
+    xpath: "<xpath_expressions>" <1>
+----
+<1> `xpath` must be a valid XPath expression.
+
+.`hasTags`
+
+The `hasTags` capability enables the provider to query application tags. It queries the internal data structure to check whether the application has the given tags.
+
+[source,terminal]
+----
+when:
+  # when more than one tags are given, a logical AND is implied
+  hasTags: <1>
+    - "tag1"
+    - "tag2"
+----
+<1> When more than one tags is given, a logical AND is implied.
+
+===== `java` provider
+
+The `java` provider analyzes Java source code.  This provider's capabilities are `referenced` and `dependency`.
+
```

Comment:
Parallel staructure with above --> This provider has the following capacities:

* `referenced`
* `dependency`